### PR TITLE
11 add features matrix class part 2

### PR DESCRIPTION
**Pull request by jonyyy1**

### CODE REVIEW

File: Google_tests/matrix/test_matrix.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Matrix<int> m3 = m1 + m2;
+    cout << m3;
+}
+
+void test_operator_minus() {
+    Matrix<int> m1({{1, 2, 3}, {4, 5, 6}, {7, 8, 9}});
+    Matrix<int> m2({{9, 8, 7}, {6, 5, 4}, {3, 2, 1}});
+    Matrix<int> m3 = m2 - m1;
+    cout << m3;
+}
```
Comment (RogerHuauya) [inline Google_tests/matrix/test_matrix.hpp:34]: Los test generalmente no tienen .hpp o h, hacer los test encapsulados en una funcion si no se va a testear una funcion directamente de una libreria

Comment (gabrielromerod) [reply Google_tests/matrix/test_matrix.hpp:34]: confirmo :+1:

File: Google_tests/matrix/test_matrix.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    EXPECT_EQ(m3(1,1), 69.0);
+
+}
+
+
+TEST(MatrixTransposeTest, Transpose){
+    Matrix<int> m1({{1, 2, 3}, {4, 5, 6}});
+    m1 = m1.t();
+    EXPECT_EQ(m1(0,0),1);
+    EXPECT_EQ(m1(1,1),5);
```
Comment (RogerHuauya) [inline Google_tests/matrix/test_matrix.cpp:130]: dont be pendex, test values that are outside of semidiagonal

File: Google_tests/matrix/test_matrix.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    Matrix<double> m2({{1.0, 2.0, 3.0}, {4.0, 5.0, 6.0}});
+    m2 = m2.t();
+    EXPECT_EQ(m2(0,0), 1.0);
+    EXPECT_EQ(m2(1,1), 5.0);
```
Comment (RogerHuauya) [inline Google_tests/matrix/test_matrix.cpp:134]: same here

**Review by RogerHuauya — CHANGES_REQUESTED** [review]
change test cases for transpose, evaluate that dimensions are transposed

File: Google_tests/matrix/test_matrix.cpp [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+TEST(MatrixTransposeTest, Transpose){
+    Matrix<int> m1({{1, 2, 3}, {4, 5, 6}});
+    m1 = m1.t();
+    EXPECT_EQ(m1(0,0),1);
+    EXPECT_EQ(m1(0,1),4);
+    EXPECT_EQ(m1(1,0),2);
+    EXPECT_EQ(m1(1,1),5);
+    EXPECT_EQ(m1(2,0),3);
+    EXPECT_EQ(m1(2,1),6);
+    EXPECT_EQ(m1.shape(), std::make_pair(3,2));
```
Comment (RogerHuauya) [inline Google_tests/matrix/test_matrix.cpp:135]: unit test for shape missing
m1(5, 1)
expect m1.shape(), makepair(5,1)

**Review by RogerHuauya — APPROVED** [review]
Lgtm

**Pull request merged**